### PR TITLE
Fix parser failures found by source alteration in test cases

### PR DESCRIPTION
**Pull request by ahoppen**
These parser failures were found by https://github.com/apple/swift-syntax/pull/1340.

- There was one round-trip failure introduced by https://github.com/apple/swift-syntax/pull/1464
- A few cases were just consume tokens as the wrong child
- The list of expected token kinds of `DeclModifierSynax` didn’t contain one modifier that was actually parsed

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

File: Tests/SwiftParserTest/translated/AvailabilityQueryUnavailabilityTests.swift [hunk 1]
```diff
@@ -473,4 +477,16 @@ final class AvailabilityQueryUnavailabilityTests: XCTestCase {
...
   }
+
+  func testAvailabilityQueryUnavailability34d() {
+    assertParse(
+      """
+      if #available(*) 1️⃣== {
+      }
+      """,
+      diagnostics: [
+        DiagnosticSpec(locationMarker: "1️⃣", message: "unexpected code '==' in 'if' statement")
```
Comment (kimdv) [inline Tests/SwiftParserTest/translated/AvailabilityQueryUnavailabilityTests.swift:488]: Maybe we should add a fixit if we start to require it in #1467 

Same for fixedSource below

**Review by kimdv — APPROVED** [review]

File: Tests/SwiftParserTest/translated/AvailabilityQueryUnavailabilityTests.swift (hunk 1 above)
Comment (ahoppen) [reply Tests/SwiftParserTest/translated/AvailabilityQueryUnavailabilityTests.swift:492]: Thanks for the reminder

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Review by bnbarham — APPROVED** [review]

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Pull request merged**